File: main/src/main/java/com/hx/main/MainActivity.java
package com.hx.main;

import android.Manifest;
import androidx.viewpager.widget.ViewPager;
import com.alibaba.android.arouter.facade.annotation.Route;
import com.hx.steven.activity.BaseActivity;
import com.hx.steven.component.ProgressBarView;
import com.hx.steven.constant.Constants;
import com.hx.steven.util.BarColorUtils;
import com.hx.steven.viewpageTransformer.ScaleInTransformer;

@Route(path = Constants.A_MAIN)
public class MainActivity extends BaseActivity {
    {
        setEnableMultiple(false);
    }

    ViewPager viewPager;
    ProgressBarView pbView;
    @Override
    protected void initView() {
        RequestPermissions(0,
                Manifest.permission.WRITE_EXTERNAL_STORAGE,
                Manifest.permission.READ_PHONE_STATE);

        /**viewPager变换操作*/
        viewPager = findViewById(R.id.id_viewpager);
        viewPager.setOffscreenPageLimit(3);
        adapter = new pageAdapter(this, images);
        viewPager.setPageTransformer(false, new ScaleInTransformer());
        viewPager.setAdapter(adapter);
        viewPager.setCurrentItem(1);
        /**自定义progressView*/
        pbView = findViewById(R.id.pbView);
        pbView.setMax(100);
        pbView.setProgress(43);
        BarColorUtils.setBarColor(this, "#C1FFC1", true);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    private pageAdapter adapter;
    private int[] images = new int[]{R.drawable.a, R.drawable.b, R.drawable.c, R.drawable.a, R.drawable.b, R.drawable.c};

    @Override
    protected int getContentId() {
        return R.layout.main_activity;
    }

}

File: main/src/main/java/com/hx/main/MainActivity.kt
package com.hx.main

import android.Manifest
import android.content.Intent
import android.view.View
import androidx.appcompat.widget.AppCompatTextView
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.alibaba.android.arouter.facade.annotation.Route
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.animation.AlphaInAnimation
import com.chad.library.adapter.base.listener.OnItemClickListener
import com.hx.steven.activity.BaseActivity
import com.hx.steven.activity.OnPermissionListener
import com.hx.steven.component.DividerItemDecoration
import com.hx.steven.constant.Constants
import com.hx.steven.mina.MinaService
import com.hx.steven.mina.SessionManager
import com.hx.steven.util.DeviceUtil
import com.orhanobut.logger.Logger
import com.tencent.mmkv.MMKV


@Route(path = Constants.A_MAIN)
class MainActivity : BaseActivity(), OnItemClickListener {
    private var recyclerView: RecyclerView? = null
    private var adapter: ExampleAdapter? = null
    private var exampleData: List<ExampleBean>? = null

    private lateinit var imeiTv: AppCompatTextView
    override fun getContentId(): Int {
        return R.layout.main_activity
    }

    override fun initView() {
        imeiTv = findViewById(R.id.main_imei)
        recyclerView = findViewById(R.id.example_recycle)
        recyclerView?.layoutManager = GridLayoutManager(this, 3)
        recyclerView?.addItemDecoration(DividerItemDecoration().setDrawOuterBorder(true))
        exampleData = listOf(
            ExampleBean("module1", R.drawable.module),
            ExampleBean("module2", R.drawable.wirehouse)
        )
        adapter = ExampleAdapter(R.layout.main_example_item, exampleData)
        adapter?.adapterAnimation = AlphaInAnimation()
        recyclerView?.adapter = adapter
        adapter?.setOnItemClickListener(this)

        RequestPermissions(
            object : OnPermissionListener {
                override fun onPermissionGranted() {
                    Logger.d("onPermissionGranted")
                    val imei = DeviceUtil.getDeviceImei(this@MainActivity, 0)
                    imeiTv.text = imei
                    MMKV.defaultMMKV().encode("imei", imei)
                }

                override fun onPermissionDenied(deniedList: MutableList<String>?) {
                    Logger.d(deniedList)
                    finish()
                }
            }, Manifest.permission.READ_PHONE_STATE
        )

        val intent = Intent(this, MinaService::class.java)
        startService(intent)
        Logger.d("connect to server")
    }

    override fun onItemClick(adapter: BaseQuickAdapter<*, *>, view: View, position: Int) {
        var name = exampleData?.get(position)?.name
        when (name) {
            "module1" -> {
                val txt = "11111"
                SessionManager.getInstance().writeToServer(txt)
                Logger.d("send message to server ：$txt")
            }
            "module2" -> {
            }
        }
    }

}